src/components/ha-card.ts

```diff
@@ -27,24 +28,46 @@ class HaCard extends LitElement {
         color: var(--primary-text-color);
         display: block;
         transition: all 0.3s ease-out;
+        position: relative;
       }
-      .header:not(:empty),
-      .header::slotted(*) {
+
+      :host ::slotted(*) {
+        display: block;
+      }
+
+      #header-slot:not(:empty), /* This matches if header/heading is set*/
+      #header-slot::slotted(*),
+      #default-slot::slotted(.card-header) {
         color: var(--ha-card-header-color, --primary-text-color);
         font-family: var(--ha-card-header-font-family, inherit);
         font-size: var(--ha-card-header-font-size, 24px);
         letter-spacing: -0.012em;
         line-height: 32px;
-        padding: 24px 16px 16px;
+        padding: 24px 16px 0 16px;
         display: block;
       }
+
+      #content-slot::slotted(*),
```

Comment:
We should not just put a padding on all things in the content slot, if the person putting things in that slot wants that, they can add `.card-content`

Reply:
This gives two alternative ways to get the same result. Either

```html
<ha-card>
  <div class="card-content">
    This is fully paper-card compatible
  </div>
  <div class="card-actions">
    A button or whatnot
  </div>
</ha-card>
```

Or
```html
<ha-card>
  <div slot="content">
    This is fully paper-card compatible
  </div>
  <div slot="actions">
    A button or whatnot
  </div>
</ha-card>
```

Reply:
But it will prevent me from using ha-card with a table that spans the full width of the card, like in this example:

![image](https://user-images.githubusercontent.com/1444314/57152438-63de3e80-6d88-11e9-869b-53d84259adad.png)

We should not do something that applies to all.

Reply:
Oh, and yes, these list items DO have padding, but they have their own padding, so that it is included in the touch target .

![image](https://user-images.githubusercontent.com/1444314/57152502-8bcda200-6d88-11e9-9c65-9a63214b0c30.png)

Reply:
Things that are slotted in the unnamed slot and not given the `.card-content` will not be formatted.

```yaml
<ha-card>
  Things with no margins.
  <div slot="actions">
    <button></button>
  </div>
</ha-card>
```

To be fair... one of the interfaces should probably go, but which one? Should there be named slots **or** classes?

Giving it two seconds of thought, my vote is probably for classes.